### PR TITLE
Switch to `pyproject.toml` and use src-layout.

### PR DESCRIPTION
**Pull request by thetorpedodog**
By using the src-layout convention and `exclude` functionality*, we can eliminate our custom `setup.py` logic entirely and switch to using a purely-declarative `pyproject.toml` file.

*: https://setuptools.pypa.io/en/latest/userguide/package_discovery.html

### CODE REVIEW

**Review by nguyenv — APPROVED** [review]

**Comment by Shelnutt2** [comment]
Does this handle if we want other namespaces besides the top level `tiledb`?

**Comment by thetorpedodog** [comment]
> Does this handle if we want other namespaces besides the top level `tiledb`?

It does, yes.

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+
 [build-system]
 requires = ["setuptools>=42", "wheel", "setuptools_scm>=6"]
 
+[tool.setuptools]
+zip-safe = false
+
+[tool.setuptools.packages.find]
+where = ["src"]
+exclude = ["tiledb"]
```
Comment (ihnorton) [inline pyproject.toml:38]: Why the exclude here?

Comment (thetorpedodog) [reply pyproject.toml:38]: I could have sworn this was necessary to make it not clobber `tiledb/__init__.py`, but apparently not? No complaints about removing it from me.

**Review by ihnorton — APPROVED** [review]

**Pull request merged**